examples/dict.roc

```diff
@@ -0,0 +1,25 @@
+app [main!] { 
+    cli: platform "https://github.com/roc-lang/basic-cli/releases/download/0.19.0/Hj-J_zxz7V9YurCSTFcFdu6cQJie4guzsPMUi5kBYUk.tar.br",
+    json: "../package/main.roc", # use release URL (ends in tar.br) for local example, see github.com/lukewilliamboswell/roc-json/releases
+}
+
+import cli.Stdout
+import json.Json
+
+main! = |_args|
+    dict = 
+        Decode.from_bytes(json_dict, Json.utf8)
+        |> Result.map_ok(Dict.from_list)
+        |> Result.with_default(Dict.empty({}))
```

Comment:
We should avoid `with_default` in examples to discourage users from writing code that fails silently.